docs/source/conceptual-guides/shadowing.md

```diff
@@ -0,0 +1,113 @@
+---
+myst:
+  html_meta:
+    "description": "How to shadow a component or module in Volto"
+    "property=og:description": "How to shadow a component or module in Volto"
+    "property=og:title": "Shadow a component or module in Volto"
+    "keywords": "Volto, shadow, override, component, frontend"
+---
+
+# Shadow a component or module
+
+Component or module shadowing allows you to override a component from an existing package in a clean and structured way, without directly modifying third-party code.
+This technique is particularly useful for customizing the behavior or appearance of specific components in Volto or any add-ons.
+
+This mechanism is powered by Webpack's [`resolve.alias`](https://webpack.js.org/configuration/resolve/#resolvealias), which maps a module path to an alternative file or folder.
+Webpack resolves the override at build time, replacing the original module with your custom implementation.
+
+To shadow a component, perform the following tasks.
+
+-   Locate the component you want to override or shadow.
+-   Replicate the component's original folder structure inside a {file}`src/customizations` folder in your add-on.
+
+
+## Folders and files hierarchy structure pattern for shadowing
+
+To shadow components, you must follow a file structure pattern.
+
+
+### Customize Volto core components or modules
+
+Assume the component you want to shadow is located in Volto core at {file}`node_modules/@plone/volto/src/components/theme/Logo/Logo.svg`.
+
+To customize a Volto core component, the first folder must be {file}`volto`.
+Replicate this structure under {file}`src/customizations` as shown.
+
+```text
+customizations/
+└── volto/
+    └── components/
+        └── theme/
+            └── Logo/
+                └── Logo.svg
+```
+
+```{note}
+Please note the absence of the {file}`src/` folder in the shadowed module paths.
+This is because the build ignores and removes the {file}`src/` prefix when resolving modules.
+```
+
+```{tip}
+Use absolute imports in your shadowed modules to avoid resolution issues.
+```
+
+
+### Customize add-ons components or modules
+
+Assume the component you want to shadow is located in Volto core at {file}`node_modules/@kitconcept/volto-light-theme/src/components/Header/Header.tsx`.
```

Comment:
This is not Volto core, should say "is located in an add-on" or something like that.